packages/rules-typescript/src/unused-dependency-detector.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  asBollLineNumber,
+  Failure,
+  FileContext,
+  Result,
+  Success,
+  PackageMetaRule,
+  asBollFile,
+  Package
+} from "@boll/core";
+import { ImportDeclaration, isImportDeclaration, isStringLiteral, SourceFile } from "typescript";
+
+const ruleName = "UnusedDependencyDetector";
+const exclude = ["@types"];
+
+export interface UnusedDependencyDetectorOptions {
+  exclude: string[];
+  ignoreDevDependencies: boolean;
+  packageContextOverride: Package;
+}
+
+interface DependencyCount {
+  [dependency: string]: number;
+}
+
+/**
+ * UnusedDependencyDetector detects packages declared as dependencies
+ * or devDependencies that are not imported anywhere in code.
+ */
+export class UnusedDependencyDetector implements PackageMetaRule {
+  private _exclude: Set<string>;
+
+  constructor(private options?: Partial<UnusedDependencyDetectorOptions>) {
+    this._exclude = options && options.exclude ? new Set([...exclude, ...options.exclude]) : new Set(exclude);
+  }
+
+  get name(): string {
+    return ruleName;
+  }
+
+  async check(files: FileContext[]): Promise<Result[]> {
+    return this.checkPackageImports(
+      files.map(f => this.getFilePackageImports(f.source)),
+      files
+    );
+  }
+
+  getFilePackageImports(sourceFile: SourceFile): string[] {
+    const importPaths: string[] = [];
+    sourceFile.forEachChild(n => {
+      if (isImportDeclaration(n)) {
+        const path = this.getPathFromNode(n);
+        if (!path.startsWith(".")) {
+          importPaths.push(path);
+        }
+      }
+    });
+    return importPaths;
+  }
+
+  checkPackageImports(packageImports: string[][], files: FileContext[]): Result[] {
+    const { dependencies, devDependencies } = this.initializeDependencyCounts(files);
+    packageImports.forEach(fi => {
+      fi.forEach(i => {
+        const packageName = this.getPackageNameFromImportPath(i);
+        dependencies[packageName] !== undefined && dependencies[packageName]++;
+        devDependencies[packageName] !== undefined && devDependencies[packageName]++;
+      });
+    });
+
+    const results: Result[] = [
+      ...Object.keys(dependencies)
+        .filter(d => dependencies[d] === 0 && !this.isExcluded(d))
+        .map(
+          d =>
+            new Failure(
+              ruleName,
+              asBollFile("package.json"),
+              asBollLineNumber(0),
```

Comment:
Shouldn't we be showing the exact line number for missing dependency instead of `0`?